lib/jekyll/page.rb

```diff
@@ -123,12 +123,10 @@ def process(name)
     # layouts      - The Hash of {"name" => "layout"}.
     # site_payload - The site payload Hash.
     #
-    # Returns nothing.
+    # Returns String rendered page.
     def render(layouts, site_payload)
-      site_payload["page"] = to_liquid
-      site_payload["paginator"] = pager.to_liquid
-
-      do_layout(site_payload, layouts)
+      site.layouts = layouts
+      Renderer.new(site, self, site_payload).run
```

Comment:
@benbalter is doing similar change in this method [here](https://github.com/jekyll/jekyll/pull/5211/files#diff-6257bc8ccdf5697787354024f8d97a78R77), but skipping `layouts` variable.
Should we do the same here?